homeassistant/components/lamarzocco/config_flow.py

```diff
@@ -282,6 +283,14 @@ async def async_step_dhcp(
     ) -> ConfigFlowResult:
         """Handle discovery via dhcp."""
 
+        # guard against hostname changes
+        if not re.match(r"^(LM|MR|GS)\d{6}$", discovery_info.hostname):
+            _LOGGER.debug(
+                "Discovered La Marzocco machine with invalid hostname: %s, most likely hostname has been changed",
+                discovery_info.hostname,
+            )
+            return self.async_abort(reason="non_serial_hostname")
+
```

Comment:
IMO there is a deeper problem here.
The unique_id should not come from the parsing of a serial number from a hostname.

It may be better to use the MAC as a unique_id, or to use `abort_entries_match`

Reply:
this was the only way to ensure uniqueness across Bluetooth discovery and manual configuration. The default hostname is the machine's hardware serial. This is an edge case anyways imo, as someone would need to have their machine in HA first and then force a new hostname through a firewall or similar. 

`abort_entries_match` is a good idea though, switched to that.